### ISSUE TITLE
c8y-firmware-plugin has dependency on tedge-agent but not declared

### ISSUE THREAD

**Issue by rina23q** [thin-edge/thin-edge.io#1793]
**Describe the bug**
`tedge-agent --init` creates a directory `/var/tedge` in their `postinst` script, and `c8y-firmware-plugin --init` has an expectation that `/var/tedge` directory already exists because it needs to create directories `/var/tedge/cache`, `/var/tedge/file-transfer`, and `/var/tedge/firmware`.

However, the dependency on `tedge-agent` is not declared in the `c8y-firmware-plugin` package, the installation may fail.
Also, since the plugin needs a downloader from `tedge-agent`, so in the view point of feature, `c8y-firmware-plugin` has really a dependency on `tedge-agent`.

_Originally posted by @reubenmiller in https://github.com/thin-edge/thin-edge.io/pull/1787#discussion_r1127793208_
            

**To Reproduce**
Run
```
    apt-get install -y \
        tedge \
        tedge-mapper \
        tedge-agent \
        tedge-apt-plugin \
        c8y-configuration-plugin \
        c8y-log-plugin \
        c8y-firmware-plugin \
        tedge-watchdog
```

Then, you will see the error that installation of c8y-firmware-plugin failed.

**Expected behavior**
The command above finishes without an error and return code must be 0.

**Screenshots**

**Environment (please complete the following information):**
 - OS [incl. version]
 - Hardware [incl. revision]
 - System-Architecture [e.g. result of "uname -a"]
 - thin-edge.io version [e.g. 0.1.0]

**Additional context**
Add a dependency on `tedge-agent` to `c8y-firmware-plugin`.
Make sure for `c8y-configuration-plugin` as well.

**Comment by rina23q** [comment]
Fixed by #1800. Also, system test in Robot Framework is added by #1800.

**Comment by gligorisaev** [comment]
Ckecked and it is OK
